TestSettingFunction fails on Py 3.8, 'a_a'[1] no longer rejected

Test case fails:
```
[   59s] setting_function_bad = <UM.Settings.SettingFunction (0x7f5891e6fc70) ='a_a'[1] >
[   59s] 
[   59s]     def test_init_bad(setting_function_bad):
[   59s]         assert setting_function_bad is not None
[   59s] >       assert not setting_function_bad.isValid()
[   59s] E       AssertionError: assert not True
[   59s] E        +  where True = <bound method SettingFunction.isValid of <UM.Settings.SettingFunction (0x7f5891e6fc70) ='a_a'[1] >>()
[   59s] E        +    where <bound method SettingFunction.isValid of <UM.Settings.SettingFunction (0x7f5891e6fc70) ='a_a'[1] >> = <UM.Settings.SettingFunction (0x7f5891e6fc70) ='a_a'[1] >.isValid
[   59s] 
[   59s] tests/Settings/TestSettingFunction.py:72: AssertionError
[   59s] =============================== warnings summary ===============================
```
Possibly elated: https://github.com/Ultimaker/Uranium/pull/592

--- a/UM/Settings/SettingFunction.py
+++ b/UM/Settings/SettingFunction.py
@@ -245,7 +245,7 @@ class _SettingExpressionVisitor(ast.NodeVisitor):
     def visit_Subscript(self, node: ast.Index):
         if type(node.value) == ast.Str:
             raise IllegalMethodError("Indexing on strings is not allowed")
-        if type(node.value) == ast.Constant and isinstance(node.value.value, str):
+        if hasattr(ast, "Constant") and type(node.value) == ast.Constant and isinstance(node.value.value, str):
             raise IllegalMethodError("Indexing on strings is not allowed")
         for child_node in ast.iter_child_nodes(node):
             self.visit(child_node)

--- a/UM/Settings/SettingFunction.py
+++ b/UM/Settings/SettingFunction.py
@@ -245,6 +245,8 @@ class _SettingExpressionVisitor(ast.NodeVisitor):
     def visit_Subscript(self, node: ast.Index):
         if type(node.value) == ast.Str:
             raise IllegalMethodError("Indexing on strings is not allowed")
+        if type(node.value) == ast.Constant and isinstance(node.value.value, str):
+            raise IllegalMethodError("Indexing on strings is not allowed")
         for child_node in ast.iter_child_nodes(node):
             self.visit(child_node)
 

